pom.xml

```diff
@@ -279,6 +277,14 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.10.1</version>
+                <configuration>
+                    <release>17</release>
```

Comment:
I would like if this remains a property. It's easier to identify at the very top of the pom.

Reply:
In the first approach, I simply replaced
```xml
<maven.compiler.source>11</maven.compiler.source>
<maven.compiler.target>11</maven.compiler.target>
```
with
```xml
<maven.compiler.release>17</maven.compiler.release>
```
but for some reason IntelliJ had issues with it and assumed all modules were at Java 5.

I will give it a try once more.

Reply:
So the problem was not about the new property itself, but with no `maven-compiler-plugin` specified in pom.xml. This resulted in version 3.1 being used, whereas only versions 3.6+ can read this new property. I added the latest `maven-compiler-plugin` to pom.xml and now everything compiles at level 17.